lib/util.js

```diff
@@ -23,9 +23,22 @@ function isDefined (v) {
   return typeof v !== 'undefined'
 }
 
+function promisify () {
+  var resolve_, reject_
+  var promise = new Promise(function (resolve, reject) {
+    [resolve_, reject_] = [resolve, reject]
+  })
+  function callback (err, value) {
+    if (err) reject_(err)
+    else resolve_(value)
+  }
+  return [ callback, promise ]
+}
```

Comment:
How about:

```js
function promisify () {
  var callback
  var promise = new Promise(function (resolve, reject) {
    callback = function callback (err, value) {
      if (err) reject(err)
      else resolve(value)
    }
  })
  
  return [ callback, promise ]
}
```

Reply:
great idea :)

Reply:
I'm curious how this _actually_ works. Is it because of `function callback () {}`? `promisify()` returns `[ undefined, new Promise() ]` right? And once `promise.then()` is called by the caller, then the Promise-callback sets `callback` to `function callback () {}`.

Reply:
@ralphtheninja it works because the `function (resolve, reject) {}` is called immediately by the Promise constructor, before the constructor even returns.

Reply:
@vweevers Thanks for the lesson! :) I always thought that `.then()` etc actually triggered the promise. TIL.